Updated the code for the Napp components, added Pinia, and added Axios

HeriLFIU opened the pull request:
Closes #166 
Closes #172 
Closes #174
Closes #173 

### Summary

Replaced `jquery` with `pinia` and `axios`.
Cleaned up code.

### Local Tests

The Napp components worked with no errors.

### End-to-End Tests

HeriLFIU commented:
@rmotitsuki These are the pinia changes

HeriLFIU commented:
@viniarck I think this is good for review and ill just do a separate PR for the k-toolbar unit tests that branch off of this one.

HeriLFIU commented:
@rmotitsuki @viniarck That big options object looks hideous, and the same one is shared between all the Napp components, should I try and move it within the store or is it ok as is?

HeriLFIU commented:
let me try it out, just with the first to see

HeriLFIU commented:
I just moved into the store, and I got no errors.
Since the 3 napp components use the same options I can also just access them through the store and make it look much cleaner.

HeriLFIU commented:
but ill do that once I get to their unit tests

HeriLFIU commented:
I tried to also move the loading of each component into the store since its shared by all 3 components, but the vue3-sfc-loader got very angry.

HeriLFIU commented:
Best not to touch it.

viniarck commented:
> Best not to touch it.

@HeriLFIU, what's the final verdict? Did you hit issues with `vue3-sfc-loader` during this refactoring?

Let me and rogerio know if we should still review this PR or if you're still working on it (if you are mark it as a draft)

HeriLFIU commented:
@HeriLFIU Oh no, that was just when I had tried to move the vue3-sfc-loader within the store. Since all 3 napp components share very similar code for it. But the current code is all good, I just left the vue3-sfc-loader as it was.

viniarck reviewed (commented):
Nicely done @HeriLFIU, +1 from me, let's see @rmotitsuki feedback too. And, before landing this, let's also ensure that when replacing ajax with axios in one of our NApps the error handling will be smooth and not too much extra work. Can you also prototype it and map a new issue with bullets for each NApp? Just so we can keep track of the ones remaining too? I don't expect surprises but let's get that effort mapped and on our radar too. Once we also got one NApp, then it'll serve as a reference of how axios and how its error handling is supposed to look like, try catch is pretty clean and readable and we can stick with it to also avoid multiple styles in our code base.

Good to see `pinia` introduced and how it was used.

viniarck commented on package.json:
```diff
@@ -23,10 +23,12 @@
     "@fortawesome/free-solid-svg-icons": "^6.5.2",
     "@fortawesome/vue-fontawesome": "^3.0.6",
     "@trevoreyre/autocomplete-vue": "^3.0.3",
+    "axios": "^1.8.4",
```
Should we go for [1.9.0](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md#190-2025-04-24)? 

Looks like its focus was mostly bug fixes

HeriLFIU replied:
yes, ill move it up and push the new changes

HeriLFIU commented on src/components/kytos/napp/Toolbar.vue:
```diff
@@ -11,110 +11,43 @@
+      try {
+        const response = await axios.get(this.url);
+        this.toolbarItems.push(...response.data)
+      } catch (err) {
+        console.error(err)
+      } finally {
```
```suggestion
      } catch (err) {
        console.error(err)
      } finally {
```
@viniarck cant we just add any error detection with the catch block?

HeriLFIU replied:
ill also prototype it for the one of the napps and create an issue for all of the ones that will need changing.

HeriLFIU replied:
Ill tell you if it functions well once Im done

viniarck replied:
> @viniarck cant we just add any error detection with the catch block?

Yes @HeriLFIU. A try catch works as you'd expect in a async context. In our code base we also have many sync funcs https://github.com/kytos-ng/mef_eline/blob/master/ui/k-info-panel/show_circuit.kytos#L381-L399, and currently as you know with jquery we've been using `request.done` and `request.fail` function callbacks. 

Maybe we should stick with `axios.then().catch().`? Since that will work for both async and sync funcs? https://axios-http.com/docs/handling_errors

Think about it and do some experiments, then we can stick with it. It's fine to mix both, but if one pattern suffices for the typical use cases we have, then it's easier to have that too since it makes easier to reason and understand, especially since production code is usually read many more times than it's written.

HeriLFIU commented:
@viniarck I fixed the race conditions by adding keys to the `k-toolbar-items` of each components. The keys are the new name attribute on the related PRs. They use the new attribute to access the correct data each time.
The changes were so small that I didn't know if I should have created an issue for them and updated the `CHANGELOG.rst`. Should I have updated them?

viniarck commented:
> @viniarck I fixed the race conditions by adding keys to the `k-toolbar-items` of each components. The keys are the new name attribute on the related PRs. They use the new attribute to access the correct data each time. The changes were so small that I didn't know if I should have created an issue for them and updated the `CHANGELOG.rst`. Should I have updated them?

@HeriLFIU, that's fine to me. Regarding the changelog, since this is just internal refactoring, and you linked back this PR on NApps PR I think it's fine, no need to update the changelog there in this case.

Regarding the rest of the review I'll leave to @rmotitsuki to approve this PR and the linked ones. Other than that, I don't have any other feedback to add to the review.

viniarck commented:
> > @viniarck I fixed the race conditions by adding keys to the `k-toolbar-items` of each components. The keys are the new name attribute on the related PRs. They use the new attribute to access the correct data each time. The changes were so small that I didn't know if I should have created an issue for them and updated the `CHANGELOG.rst`. Should I have updated them?
> 
> @HeriLFIU, that's fine to me. Regarding the changelog, since this is just internal refactoring, and you linked back this PR on NApps PR I think it's fine, no need to update the changelog there in this case.
> 
> Regarding the rest of the review I'll leave to @rmotitsuki to approve this PR and the linked ones. Other than that, I don't have any other feedback to add to the review.

The only related point I still have is the one [I raised on this comment](https://github.com/kytos-ng/mef_eline/pull/659#discussion_r2069033941). It's up to you if you'll try to expose that helper func on this PR here or another one.

HeriLFIU commented:
@viniarck I did it on another PR. 
https://github.com/kytos-ng/ui/pull/176

HeriLFIU commented:
@rmotitsuki @viniarck I have done what I originally set out to do. I moved the component registration logic to the store, and now I can do it for all of the Napp components, and remove a ton of duplicated code.

HeriLFIU commented:
@viniarck @rmotitsuki I just made the changes across all of the Napp components. It should all be good.

viniarck reviewed (approved):
Nicely done @HeriLFIU, and thanks @rmotitsuki for the review. 

Let's go ahead and ship this, go ahead and merge it, also let the team know on Slack dev channel if someone is using `master`, to rebuild the `ui` since new deps were introduced.
